tests/pyminisketch.py

```diff
@@ -0,0 +1,514 @@
+    """Test class for poly_find_roots."""
+
+    def field_size_test(self, field_size):
+        """Run tests for given field_size."""
+        gf = GF2Ops(field_size)
+        for test_size in [0, 1, 2, 3, 10]:
+            roots = [random.randrange(1 << field_size) for _ in range(test_size)]
+            roots_set = set(roots)
+            # Construct a polynomial with all elements of roots as roots (with multiplicity).
+            poly = [1]
+            for root in roots:
+                new_poly = [0] + poly
+                for n, c in enumerate(poly):
+                    new_poly[n] ^= gf.mul(c, root)
+                poly = new_poly
+            # Invoke the root finding algorithm.
+            found_roots = poly_find_roots(poly, gf)
+            # The result must match the input, unless any roots were repeated.
+            if len(roots) == len(roots_set):
+                self.assertEqual(found_roots, sorted(roots))
+            else:
+                self.assertEqual(found_roots, [])
+
+    def test(self):
+        """Run tests."""
+        for field_size in range(2, 8):
+            self.field_size_test(field_size)
+
+def berlekamp_massey(syndromes, gf):
+    """Implement the Berlekamp-Massey algorithm.
+
+    Takes as input a sequence of GF(2^field_size) elements, and returns the shortest LSFR
+    that generates it, represented as a polynomial.
+    """
+    # See https://en.wikipedia.org/wiki/Berlekamp%E2%80%93Massey_algorithm.
+    current = [1]
+    prev = [1]
+    b_inv = 1
+    for n, discrepancy in enumerate(syndromes):
+        # Compute discrepancy
+        for i in range(1, len(current)):
+            discrepancy ^= gf.mul(syndromes[n - i], current[i])
+
+        # Correct if discrepancy is nonzero.
+        if discrepancy:
+            x = n + 1 - (len(current) - 1) - (len(prev) - 1)
+            if 2 * (len(current) - 1) <= n:
+                tmp = list(current)
+                current.extend(0 for _ in range(len(prev) + x - len(current)))
+                mul = gf.mul(discrepancy, b_inv)
+                for i, v in enumerate(prev):
+                    current[i + x] ^= gf.mul(mul, v)
+                prev = tmp
+                b_inv = gf.inv(discrepancy)
+            else:
+                mul = gf.mul(discrepancy, b_inv)
+                for i, v in enumerate(prev):
+                    current[i + x] ^= gf.mul(mul, v)
+    return current
+
+class Minisketch:
+    """A Minisketch sketch.
+
+    This represents a sketch of a certain capacity, with elements of a certain bit size.
+    """
+
+    def __init__(self, field_size, capacity):
+        """Initialize an empty sketch with the specified field_size size and capacity."""
+        self.field_size = field_size
+        self.capacity = capacity
+        self._odd_syndromes = [0] * capacity
+        self._gf = GF2Ops(field_size)
+
+    def add(self, element):
+        """Add an element to this sketch. 1 <= element < 1**field_size."""
```

Comment:
nit fixup per @glozow comment in https://bitcoincore.reviews/minisketch-26#l-152
```suggestion
        """Add an element to this sketch. 1 <= element < 2**field_size."""
```

Reply:
Fixed.